README.md

```diff
@@ -33,69 +47,225 @@
+    group by stop_id
+)
+select
+    stop_name,
+    estimated_pop_800m,
+    the_geom
+from septa_bus_stop_surrounding_population
+join septa_bus_stops using (stop_id)
+order by estimated_pop_800m asc
+limit 1
   )
   ```
 
-3. Using the Philadelphia Water Department Stormwater Billing Parcels dataset, pair each parcel with its closest bus stop. The final result should give the parcel address, bus stop name, and distance apart in meters. Order by distance (largest on top).
-
+@@ -41,9 +115,39 @@
   **Structure:**
   ```sql
   (
       address text,  -- The address of the parcel
       stop_name text,  -- The name of the bus stop
       distance_m double precision  -- The distance apart in meters
+      -- create a geometry column on buses
+alter table septa_bus_stops
+    add column if not exists geometry geometry(Point, 4326);
+update septa_bus_stops
+  set geometry = ST_SetSRID(ST_MakePoint(stop_lon, stop_lat), 4326);
+  
+-- indexes
+create index if not exists septa_bus_stops__the_geom__32129__idx
+    on septa_bus_stops
+    using GiST (ST_Transform(geometry, 32129));
+create index if not exists phl_pwd_parcels_geometry_32129_idx
+    on phl_pwd_parcels
+    using GiST (ST_Transform(geometry, 32129));
+-- query & cross join
+SELECT 
+    p.address,
+    cl.stop_name,
+    cl.distance_m
+from phl_pwd_parcels as p
+cross join lateral(
+    select 
+        s.stop_name,
+        ST_Distance(
+            ST_Transform(p.geometry, 32129),
+            ST_Transform(s.geometry, 32129)
+        )as distance_m
+    from septa_bus_stops as s
+    order by distance_m desc
+    limit 1
+) as cl;
   )
   ```
 
-4. Using the _shapes.txt_ file from GTFS bus feed, find the **two** routes with the longest trips. In the final query, give the `trip_headsign` that corresponds to the `shape_id` of this route and the length of the trip.
-
+@@ -52,26 +156,107 @@
   **Structure:**
   ```sql
   (
       trip_headsign text,  -- Headsign of the trip
       trip_length double precision  -- Length of the trip in meters
+      --Min in Sequence
+WITH min_t as (
+select shape_id, min(CAST(shape_pt_sequence as INT)) min_station
+from septa_bus_shapes
+GROUP BY shape_id
+),
+--Max in Sequence
+max_t as (
+select shape_id, max(CAST(shape_pt_sequence as INT)) max_station
+from septa_bus_shapes
+GROUP BY shape_id
+),
+-- Lat/Long
+septa_origin as (
+SELECT shape_id, shape_pt_lon, shape_pt_lat, CAST(shape_pt_sequence as INT) sequence_int
+FROM septa_bus_shapes
+),
+--Makeline, First in Sequence
+first as(
+SELECT ST_Makeline(ST_setsrid(ST_MakePoint(o.shape_pt_lon, o.shape_pt_lat), 4326) ORDER BY sequence_int)::geometry geom, o.shape_id
+FROM septa_origin o
+GROUP BY o.shape_id),
+--Makeline, Last in Sequence
+last as(
+SELECT ST_Makeline(ST_setsrid(ST_MakePoint(o.shape_pt_lon, o.shape_pt_lat),4326) ORDER BY sequence_int)::geometry geom, o.shape_id
+FROM septa_origin o
+GROUP BY o.shape_id)
+--Return 2 Longest Trips
+SELECT *, (ST_Length(geom)) FROM first
```

Comment:
This is yields the correct lengths _if_ you transform the geometry (the line) into a distance-based CRS (e.g., 32129); you want to avoid measuring distances in lat/lng.